src/krb5_plugin/sssd_krb5_locator_plugin.c

```diff
@@ -546,7 +559,8 @@ krb5_error_code sssd_krb5_locator_lookup(void *private_data,
         return KRB5_PLUGIN_NO_HANDLE;
 
     for (c = 0; addr[c].addr != NULL; c++) {
```

Comment:
Not related to this patch, but I wonder if it would be a good idea to check the validity of addr before dereferencing it to prevent from a bug crashing an application?

Reply:
Good idea, I added a check to the latest version.